### DISCUSSION TITLE
xapian produces lots of InvalidArgumentError - Term too long

### DISCUSSION THREAD

**Discussion by tve** [docker-mailserver/docker-mailserver#4203] [Q&A]
I just upgraded to 14.0 and decided to try fts-xapian. The indexing produces a lot of Term too long errors and then spends time retrying:
```
mail    | 2024-09-28T21:23:58.754852-07:00 mail xapian-docswriter: DW #162 (Rxxxxx.Out-13,/var/mail/example.com/tv
e/xapian-indexes/db_21617c27db315257543800007409c414) - Can't add document3 : InvalidArgumentError - Term too long (> 24
5): XBDY                                                                                                                
             (null)
mail    | 2024-09-28T21:23:58.795488-07:00 mail xapian-docswriter: DW #162 (Rxxxxx.Out-13,/var/mail/example.com/tv
e/xapian-indexes/db_21617c27db315257543800007409c414) -  Retrying (/var/mail/example.com/tve/xapian-indexes/db_21617c2
7db315257543800007409c414) from 
```
One solution I found is https://stackoverflow.com/questions/30925487/xapian-search-terms-which-exceed-the-245-character-length-invalidargumenterror but I have no idea where I would put that. Am I the only one who hits this issue?

**Comment by polarathene** [comment]
Can you please confirm if the issue still occurs in `:edge` release? We did fix another xapian bug by an updated release: https://github.com/docker-mailserver/docker-mailserver/pull/4095

> but I have no idea where I would put that

That looks like it's Python? That's not a solution we could use since the project itself is C?: https://github.com/grosjo/fts-xapian

There was a report of this error back in 2020 at the projects github: https://github.com/grosjo/fts-xapian/issues/33#issuecomment-580021802

You can see that they "fixed" it by changing from error to warning. No word should be that long to index as they pointed out so it's not something useful to index anyway.

---

We could probably include another ignore rule for `logwatch` related to this, but I'm not sure how it'd target the related 2nd line. `:edge` added some additional ignore rules here: https://github.com/docker-mailserver/docker-mailserver/pull/4060

That's only relevant to `logwatch` not general log output filtering though.
